(window.webpackJsonp=window.webpackJsonp||[]).push([[7],{375:function(e,a,t){e.exports=t.p+"assets/img/VISOR.db778a43.png"},388:function(e,a,t){"use strict";t.r(a);var n=t(45),i=Object(n.a)({},(function(){var e=this,a=e.$createElement,n=e._self._c||a;return n("ContentSlotsDistributor",{attrs:{"slot-key":e.$parent.slotKey}},[n("p",[n("img",{attrs:{src:t(375),alt:"VISOR logo"}})]),e._v(" "),n("p",[n("a",{attrs:{href:"https://github.com/davidebolo1993/VISOR",target:"_blank",rel:"noopener noreferrer"}},[e._v("VISOR"),n("OutboundLink")],1),e._v(" is an efficient and versatile command-line application capable to simulate structural variants and small/single-nucleotide variants in a haplotype-resolved manner. VISOR currently supports:")]),e._v(" "),n("ol",[n("li",[e._v("simulations of short (Illumina) and long (PacBio-ONT) reads sequencing data")]),e._v(" "),n("li",[e._v("simulations of bulk and strand-seq sequencing data")]),e._v(" "),n("li",[e._v("simulations of 10X linked reads sequencing data (BETA release)")])]),e._v(" "),n("p",[e._v("VISOR is readily applicable to canger genomics, enabling the simulation of tumour purity (normal in tumour contamination), heterogeneity (mix of several subclones) and aneuploidy. VISOR also incorporates capture biases, a crucial feature for whole-exome data sets and panel sequencing applications.\nAll code is open-source and hosted on "),n("a",{attrs:{href:"https://github.com/davidebolo1993/VISOR",target:"_blank",rel:"noopener noreferrer"}},[e._v("VISOR's GitHub page"),n("OutboundLink")],1),e._v(".")]),e._v(" "),n("p",[e._v("Contents:")]),e._v(" "),n("ol",[n("li",[n("RouterLink",{attrs:{to:"/installation/installation.html"}},[e._v("Installation")])],1),e._v(" "),n("li",[n("RouterLink",{attrs:{to:"/usage/usage.html"}},[e._v("General usage")])],1),e._v(" "),n("li",[n("RouterLink",{attrs:{to:"/usecases/usecases.html"}},[e._v("Use cases")])],1)]),e._v(" "),n("div",{staticClass:"custom-block tip"},[n("p",{staticClass:"custom-block-title"},[e._v("TIP")]),e._v(" "),n("p",[e._v("For questions, help or feature requests, please open an "),n("a",{attrs:{href:"https://github.com/davidebolo1993/VISOR/issues",target:"_blank",rel:"noopener noreferrer"}},[e._v("issue"),n("OutboundLink")],1),e._v(" or get in touch with me at davidebolognini7@gmail.com")])]),e._v(" "),n("h2",{attrs:{id:"citation"}},[n("a",{staticClass:"header-anchor",attrs:{href:"#citation"}},[e._v("#")]),e._v(" Citation")]),e._v(" "),n("p",[e._v("Are you using VISOR in your works? Please cite:")]),e._v(" "),n("blockquote",[n("p",[e._v("Davide Bolognini, Ashley Sanders, Jan O Korbel, Alberto Magi, Vladimir Benes, Tobias Rausch. "),n("a",{attrs:{href:"https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btz719/5582674/",target:"_blank",rel:"noopener noreferrer"}},[e._v("VISOR: a versatile haplotype-aware structural variant simulator for short and long read sequencing"),n("OutboundLink")],1),e._v(". Bioinformatics. 2019 Oct 7.")])]),e._v(" "),n("h2",{attrs:{id:"related-works"}},[n("a",{staticClass:"header-anchor",attrs:{href:"#related-works"}},[e._v("#")]),e._v(" Related works")]),e._v(" "),n("p",[e._v("VISOR has been applied to simulate Strand-seq data in:")]),e._v(" "),n("blockquote",[n("p",[e._v("Ashley D Sanders, Sascha Meiers, Maryam Ghareghani, David Porubsky, Hyobin Jeong, M Alexandra CC van Vliet, Tobias Rausch, Paulina Richter-Pechańska, Joachim B Kunz, Silvia Jenni, Davide Bolognini, Gabriel MC Longo, Benjamin Raeder, Venla Kinanen, Jürgen Zimmermann, Vladimir Benes, Martin Schrappe, Balca R Mardin, Andreas E Kulozik, Beat Bornhauser, Jean-Pierre Bourquin, Tobias Marschall, Jan O Korbel. "),n("a",{attrs:{href:"https://www.nature.com/articles/s41587-019-0366-x",target:"_blank",rel:"noopener noreferrer"}},[e._v("Single-cell analysis of structural variations and complex rearrangements with tri-channel processing"),n("OutboundLink")],1),e._v(". Nature Biotechnology. 2019 Dec 23.")])]),e._v(" "),n("p",[e._v("VISOR has been applied to simulate repeat contractions/expansions in long-read data, as described in:")]),e._v(" "),n("blockquote",[n("p",[e._v("Davide Bolognini, Alberto Magi, Vladimir Benes, Jan O Korbel, Tobias Rausch. "),n("a",{attrs:{href:"https://academic.oup.com/gigascience/article/9/10/giaa101/5918863?searchresult=1",target:"_blank",rel:"noopener noreferrer"}},[e._v("TRiCoLOR: tandem repeat profiling using whole-genome long-read sequencing data"),n("OutboundLink")],1),e._v(". GigaScience. 2020 Oct 7.")])]),e._v(" "),n("p",[e._v("VISOR has been applied to simulate SVs in long-read data, as described in:")]),e._v(" "),n("blockquote",[n("p",[e._v("Davide Bolognini, Alberto Magi. "),n("a",{attrs:{href:"https://doi.org/10.3389/fgene.2021.761791",target:"_blank",rel:"noopener noreferrer"}},[e._v("Evaluation of Germline Structural Variant Calling Methods for Nanopore Sequencing Data"),n("OutboundLink")],1),e._v(". Front. Genet. 2021 Nov 18.")])])])}),[],!1,null,null,null);a.default=i.exports}}]);