<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>General usage | VISOR documentation</title>
    <meta name="generator" content="VuePress 1.8.2">
    
    <meta name="description" content="">
    
    <link rel="preload" href="/visordoc/assets/css/0.styles.0ec6c61e.css" as="style"><link rel="preload" href="/visordoc/assets/js/app.6e82cc75.js" as="script"><link rel="preload" href="/visordoc/assets/js/3.6c37c2a3.js" as="script"><link rel="preload" href="/visordoc/assets/js/10.2e0a9678.js" as="script"><link rel="prefetch" href="/visordoc/assets/js/11.6a825e0e.js"><link rel="prefetch" href="/visordoc/assets/js/4.3ccffad1.js"><link rel="prefetch" href="/visordoc/assets/js/5.894a9c75.js"><link rel="prefetch" href="/visordoc/assets/js/6.651a9e2f.js"><link rel="prefetch" href="/visordoc/assets/js/7.0f6365c2.js"><link rel="prefetch" href="/visordoc/assets/js/8.5e3a002f.js"><link rel="prefetch" href="/visordoc/assets/js/9.0911e29b.js"><link rel="prefetch" href="/visordoc/assets/js/vendors~docsearch.eec9cc92.js">
    <link rel="stylesheet" href="/visordoc/assets/css/0.styles.0ec6c61e.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/visordoc/" class="home-link router-link-active"><!----> <span class="site-name">VISOR documentation</span></a> <div class="links"><form id="search-form" role="search" class="algolia-search-wrapper search-box"><input id="algolia-search-input" class="search-query"></form> <nav class="nav-links can-hide"><div class="nav-item"><a href="/visordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/visordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/visordoc/usage/usage.html" aria-current="page" class="nav-link router-link-exact-active router-link-active">
  General usage
</a></div><div class="nav-item"><a href="/visordoc/usecases/usecases.html" class="nav-link">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/VISOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/visordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/visordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/visordoc/usage/usage.html" aria-current="page" class="nav-link router-link-exact-active router-link-active">
  General usage
</a></div><div class="nav-item"><a href="/visordoc/usecases/usecases.html" class="nav-link">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/VISOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/visordoc/installation/installation.html" class="sidebar-link">Installation</a></li><li><a href="/visordoc/usage/usage.html" aria-current="page" class="active sidebar-link">General usage</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/visordoc/usage/usage.html#visor-hack" class="sidebar-link">VISOR HACk</a></li><li class="sidebar-sub-header"><a href="/visordoc/usage/usage.html#visor-shorts-and-laser" class="sidebar-link">VISOR SHORtS and LASeR</a></li><li class="sidebar-sub-header"><a href="/visordoc/usage/usage.html#visor-xenia-beta" class="sidebar-link">VISOR XENIA (BETA)</a></li></ul></li><li><a href="/visordoc/usecases/usecases.html" class="sidebar-link">Use cases</a></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="general-usage"><a href="#general-usage" class="header-anchor">#</a> General usage</h1> <p><a href="https://github.com/davidebolo1993/VISOR" target="_blank" rel="noopener noreferrer">VISOR<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> is built on 3 modules:</p> <ol><li><a href="https://github.com/davidebolo1993/VISOR/tree/master/VISOR/HACk" target="_blank" rel="noopener noreferrer">HACk<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the 'HAplotype Creator'</li> <li><a href="https://github.com/davidebolo1993/VISOR/tree/master/VISOR/SHORtS" target="_blank" rel="noopener noreferrer">SHORtS<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the 'SHOrt Reads Simulator'</li> <li><a href="https://github.com/davidebolo1993/VISOR/tree/master/VISOR/LASeR" target="_blank" rel="noopener noreferrer">LASeR<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the 'Long reAdS simulatoR'</li> <li><a href="https://github.com/davidebolo1993/VISOR/tree/master/VISOR/XENIA" target="_blank" rel="noopener noreferrer">XENIA<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>, the '10X gENomics sImulAtor' [BETA]</li></ol> <h2 id="visor-hack"><a href="#visor-hack" class="header-anchor">#</a> VISOR HACk</h2> <p>HACk allows users to insert haplotype-specific structural and small/single-nucleotide variants in a <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp" target="_blank" rel="noopener noreferrer">FASTA<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> reference template once a proper file in <a href="https://www.ensembl.org/info/website/upload/bed.html" target="_blank" rel="noopener noreferrer">BED<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>-like format describing these variants is provided.</p> <h3 id="hack-bed"><a href="#hack-bed" class="header-anchor">#</a> HACk BED</h3> <p>HACk requires a BED with 6 columns, like the one provided <a href="https://github.com/davidebolo1993/VISOR/blob/master/Examples/HACk.bed" target="_blank" rel="noopener noreferrer">here<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. Each entry describes a structural or small/single-nucleotide variant.</p> <ol><li>Column 1 contains the name of the chromosome</li> <li>Column 2 contains the start coordinate (the breakpoint-1 coordinate for insertions and de novo tandem repeat creation)</li> <li>Column 3 contains the end coordinate (the breakpoint coordinate for insertions and de novo tandem repeat creation)</li> <li>Column 4 contains the type of the variant. Users can choose between:
<ul><li>deletion (delete from start to end)</li> <li>inversion (invert from start to end)</li> <li>tandem duplication (duplicate from start to end)</li> <li>inverted tandem duplication (duplicate from start to end and invert duplicated segment)</li> <li>insertion (insert sequence immediately after end)</li> <li>tandem repeat expansion (expand a pre-existent microsatellite, with start and end coordinates specified as in <a href="https://raw.githubusercontent.com/davidebolo1993/VISOR/master/Examples/GRCh38.microsatellites.bed" target="_blank" rel="noopener noreferrer">this example<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> for human hg38 reference genome)</li> <li>tandem repeat contraction (contract a pre-existent microsatellite, as above)</li> <li>perfect tandem repetition (insert a new perfect tandem repetition immediately after end)</li> <li>approximate tandem repetition (insert a new approximate tandem repetition immediately after end)</li> <li>translocation cut-paste (translocate and delete from start to end)</li> <li>translocation copy-paste (alias 'interspersed duplication' is also accepted. Translocate from start to end)</li> <li>reciprocal translocation (translocate from start to end and replace translocated region with another)</li> <li>SNP (generate a SNP in end)</li> <li>MNP (generate multiple SNPs from start to end)</li></ul></li> <li>Column 5 contains the information for the variant. Must follow the rules below:
<ul><li>for deletion and inversion, must be None (means, no other information to add. Exceptionally, for users who want to simulate 1bp deletions, column 5 can be '1bp', which means that the end coordinate is not included in the deletion)</li> <li>for tandem duplication and inverted tandem duplication, must be an integer, specifying the number of times the region have to be duplicated</li> <li>for insertion, must be a string of valid DNA nucleotides</li> <li>for tandem repeat expansion and tandem repeat contraction, must be a string with motif to expand or contract (M) and number of motifs to add or remove (N), in the format M:N</li> <li>for perfect tandem repetition, must be a string with the repetition motif (M) and the number of motifs to insert (N), in the format M:N</li> <li>for approximate tandem repetition, must be a string with the repetition motif (M), the number of motifs to insert (N) and the number of errors (E), in the format M:N:E</li> <li>for translocation cut-paste and translocation copy-paste, must be a string with the haplotype (H), the chromosome (C), the breakpoint (B) and the orientation (O), in the format H:C:B:O. Breakpoint is the base immediately before the one where translocated region will be put</li> <li>for reciprocal translocation, must be a string with the haplotype (H), the chromosome (C), the breakpoint (B), the orientation of the first region (OA) and the orientation of the second (OB), in the format H:C:B:OA:OB. Breakpoint specification follows the rule above</li> <li>for SNP, must be a valid DNA nucleotide</li> <li>for MNP, must be a string of valid DNA nucleotides</li></ul></li> <li>Column 6 contains the information for the length of a random, non-template sequence to add at start breakpoint. If 0, non-template sequence is not added.</li></ol> <p>Once a BED is built and a reference FASTA is available, one can run HACk.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#VISOR HACk --help</span>
VISOR HACk -b <span class="token operator">&lt;</span>HACK.BED<span class="token operator">&gt;</span> -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTHACK<span class="token operator">&gt;</span>
</code></pre></div><p>For each BED provided, HACk generates a FASTA haplotype. Multiple FASTA haplotypes are generated also if a single BED is given but it contains informations about regions being translocated between different haplotypes. Use cases for HACk are described in the <a href="/visordoc/usecases/usecases.html">Use cases section</a>.</p> <h2 id="visor-shorts-and-laser"><a href="#visor-shorts-and-laser" class="header-anchor">#</a> VISOR SHORtS and LASeR</h2> <p>SHORtS and LASeR allow users to simulate Illumina paired-end short reads and Oxford Nanopore Technologies/Pacific Biosciences long reads from selected genomic regions (even entire chromosomes) once a proper file in BED-like format describing these regions, a FASTA reference template and a folder containing HACk output are provided.</p> <h3 id="shorts-and-laser-bed"><a href="#shorts-and-laser-bed" class="header-anchor">#</a> SHORtS and LASeR BED</h3> <p>SHORtS and LASeR require a BED with 5 columns, like the one provided <a href="https://github.com/davidebolo1993/VISOR/blob/master/Examples/SHORtS.LASeR.bed" target="_blank" rel="noopener noreferrer">here<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. Each entry describes a genomic region to simulate.</p> <ol><li>Column 1 contains the name of the chromosome</li> <li>Column 2 contains the start coordinate</li> <li>Column 3 contains the end coordinate</li> <li>Column 4 contains the capture bias, a float percentage describing coverage fluctuations. For example, 100.0 describes a region without coverage fluctuations, while 80.0 describes a region covered by the 80% of the simulated coverage</li> <li>Column 5 contains the purity, a float percentage describing normal in tumour contamination. For example, 100.0 describes a region without normal contamination, while 80.0 a region with 20% of the reads simulated from the corresponding normal reference</li></ol> <p>Once a BED is built, one can run SHORtS or LASeR.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#VISOR SHORtS --help</span>
VISOR SHORtS -s <span class="token operator">&lt;</span>OUTPUTHACK<span class="token operator">&gt;</span> -b <span class="token operator">&lt;</span>SHORTS.LASER.BED<span class="token operator">&gt;</span> -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTSHORTS<span class="token operator">&gt;</span>
<span class="token comment">#VISOR LASeR --help</span>
VISOR LASeR -s <span class="token operator">&lt;</span>OUTPUTHACK<span class="token operator">&gt;</span> -b <span class="token operator">&lt;</span>SHORTS.LASER.BED<span class="token operator">&gt;</span> -g <span class="token operator">&lt;</span>REFERENCE.FASTA<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTLASER<span class="token operator">&gt;</span>
</code></pre></div><p>SHORtS and LASeR store in the output folder a sorted BAM. Use cases for SHORtS and LASeR are described in the <a href="/visordoc/usecases/usecases.html">Use cases section</a>.</p> <h2 id="visor-xenia-beta"><a href="#visor-xenia-beta" class="header-anchor">#</a> VISOR XENIA (BETA)</h2> <p>XENIA currently allows users to simulate <a href="https://www.10xgenomics.com/linked-reads/" target="_blank" rel="noopener noreferrer">10x Genomics linked reads<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> from selected genomic regions (even entire chromosomes) once a proper file in standard BED format describing these regions and a folder containing HACk output are provided.</p> <h3 id="xenia-bed"><a href="#xenia-bed" class="header-anchor">#</a> XENIA BED</h3> <p>XENIA requires a standard BED with 3 columns. Each entry describes a region to simulate.</p> <ol><li>Column 1 contains the name of the chromosome</li> <li>Column 2 contains the start coordinate</li> <li>Column 3 contains the end coordinate</li></ol> <p>Once a BED is built, one can run XENIA.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#VISOR XENIA --help</span>
VISOR XENIA -s <span class="token operator">&lt;</span>OUTPUTHACK<span class="token operator">&gt;</span> -b <span class="token operator">&lt;</span>XENIA.BED<span class="token operator">&gt;</span> -o <span class="token operator">&lt;</span>OUTPUTXENIA<span class="token operator">&gt;</span>
</code></pre></div><p>Non-stadard BED with more than 3 columns (like the one described for SHORtS and LASeR) are also accepted but additional informations are simply ignored. XENIA stores in the output folder a FASTQ pair (R1-R2) for each haplotype in the HACk input folder (L001 for haplotype 1, L002 for haplotype 2, ...). FASTQ generated can be readily aligned with 10X aligner <a href="https://support.10xgenomics.com/genome-exome/software/pipelines/latest/what-is-long-ranger" target="_blank" rel="noopener noreferrer">Long Ranger<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. Use cases for XENIA are described in the <a href="/visordoc/usecases/usecases.html">Use cases section</a>.</p> <div class="custom-block tip"><p class="custom-block-title">TIP</p> <p>Please be aware that XENIA is released as a BETA version and some issues may emerge while running; if so, please open an <a href="https://github.com/davidebolo1993/VISOR/issues" target="_blank" rel="noopener noreferrer">issue<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> or get in touch with me at davidebolognini7@gmail.com</p></div></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/visordoc/installation/installation.html" class="prev">
        Installation
      </a></span> <span class="next"><a href="/visordoc/usecases/usecases.html">
        Use cases
      </a>
      →
    </span></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/visordoc/assets/js/app.6e82cc75.js" defer></script><script src="/visordoc/assets/js/3.6c37c2a3.js" defer></script><script src="/visordoc/assets/js/10.2e0a9678.js" defer></script>
  </body>
</html>
