<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Use cases | VISOR documentation</title>
    <meta name="generator" content="VuePress 1.8.2">
    
    <meta name="description" content="">
    
    <link rel="preload" href="/visordoc/assets/css/0.styles.0ec6c61e.css" as="style"><link rel="preload" href="/visordoc/assets/js/app.6e82cc75.js" as="script"><link rel="preload" href="/visordoc/assets/js/3.6c37c2a3.js" as="script"><link rel="preload" href="/visordoc/assets/js/11.6a825e0e.js" as="script"><link rel="prefetch" href="/visordoc/assets/js/10.2e0a9678.js"><link rel="prefetch" href="/visordoc/assets/js/4.3ccffad1.js"><link rel="prefetch" href="/visordoc/assets/js/5.894a9c75.js"><link rel="prefetch" href="/visordoc/assets/js/6.651a9e2f.js"><link rel="prefetch" href="/visordoc/assets/js/7.0f6365c2.js"><link rel="prefetch" href="/visordoc/assets/js/8.5e3a002f.js"><link rel="prefetch" href="/visordoc/assets/js/9.0911e29b.js"><link rel="prefetch" href="/visordoc/assets/js/vendors~docsearch.eec9cc92.js">
    <link rel="stylesheet" href="/visordoc/assets/css/0.styles.0ec6c61e.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/visordoc/" class="home-link router-link-active"><!----> <span class="site-name">VISOR documentation</span></a> <div class="links"><form id="search-form" role="search" class="algolia-search-wrapper search-box"><input id="algolia-search-input" class="search-query"></form> <nav class="nav-links can-hide"><div class="nav-item"><a href="/visordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/visordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/visordoc/usage/usage.html" class="nav-link">
  General usage
</a></div><div class="nav-item"><a href="/visordoc/usecases/usecases.html" aria-current="page" class="nav-link router-link-exact-active router-link-active">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/VISOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></div> <!----></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/visordoc/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/visordoc/installation/installation.html" class="nav-link">
  Installation
</a></div><div class="nav-item"><a href="/visordoc/usage/usage.html" class="nav-link">
  General usage
</a></div><div class="nav-item"><a href="/visordoc/usecases/usecases.html" aria-current="page" class="nav-link router-link-exact-active router-link-active">
  Use cases
</a></div><div class="nav-item"><a href="https://github.com/davidebolo1993/VISOR.git" target="_blank" rel="noopener noreferrer" class="nav-link external">
  GitHub
  <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></div> <!----></nav>  <ul class="sidebar-links"><li><a href="/visordoc/installation/installation.html" class="sidebar-link">Installation</a></li><li><a href="/visordoc/usage/usage.html" class="sidebar-link">General usage</a></li><li><a href="/visordoc/usecases/usecases.html" aria-current="page" class="active sidebar-link">Use cases</a><ul class="sidebar-sub-headers"><li class="sidebar-sub-header"><a href="/visordoc/usecases/usecases.html#visor-hack" class="sidebar-link">VISOR HACk</a></li><li class="sidebar-sub-header"><a href="/visordoc/usecases/usecases.html#visor-shorts-and-laser" class="sidebar-link">VISOR SHORtS and LASeR</a></li><li class="sidebar-sub-header"><a href="/visordoc/usecases/usecases.html#visor-xenia-beta" class="sidebar-link">VISOR XENIA (BETA)</a></li></ul></li></ul> </aside> <main class="page"> <div class="theme-default-content content__default"><h1 id="use-cases"><a href="#use-cases" class="header-anchor">#</a> Use cases</h1> <p>Following are described some use cases, examples illustrating how to run VISOR modules.</p> <h2 id="visor-hack"><a href="#visor-hack" class="header-anchor">#</a> VISOR HACk</h2> <p>As described in the <a href="/visordoc/usage/usage.html">General usage section</a>, HACk requires at least one variant file in BED format and a reference template FASTA as inputs. We can start by downloading and subsetting a reference genome.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#create the test folder</span>
<span class="token function">mkdir</span> visortest <span class="token operator">&amp;&amp;</span> <span class="token builtin class-name">cd</span> visortest
<span class="token comment">#download the human reference genome in FASTA format</span>
<span class="token function">curl</span> -LO ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/GRCh38_reference_genome/GRCh38_full_analysis_set_plus_decoy_hla.fa
<span class="token comment">#extract 2 small chromosomes</span>
samtools faidx GRCh38_full_analysis_set_plus_decoy_hla.fa chr21 <span class="token operator">&gt;</span> chr21.fa
samtools faidx GRCh38_full_analysis_set_plus_decoy_hla.fa chr22 <span class="token operator">&gt;</span> chr22.fa
<span class="token comment">#combine the 2 chromosomes in a small reference</span>
<span class="token function">cat</span> chr21.fa chr22.fa <span class="token operator">&gt;</span> small.reference.fa
samtools faidx small.reference.fa
<span class="token comment">#get a BED with regions we want to exclude when creating variants</span>
<span class="token function">curl</span> -LO https://gist.githubusercontent.com/chapmanb/4c40f961b3ac0a4a22fd/raw/2025f3912a477edc597e61d911bd1044dc943440/sv_repeat_telomere_centromere.bed
<span class="token builtin class-name">echo</span> -e <span class="token string">&quot;chr21<span class="token entity" title="\n">\n</span>chr22&quot;</span> <span class="token operator">&gt;</span> tmp.txt <span class="token operator">&amp;&amp;</span> <span class="token function">grep</span> -w -f tmp.txt sv_repeat_telomere_centromere.bed <span class="token operator">|</span> sortBed <span class="token operator">&gt;</span> exclude.bed <span class="token operator">&amp;&amp;</span> <span class="token function">rm</span> tmp.txt
</code></pre></div><h3 id="manually-building-hack-bed"><a href="#manually-building-hack-bed" class="header-anchor">#</a> Manually building HACk BED</h3> <p>For a limited number of variants, manually building HACk BED as described in the <a href="/visordoc/usage/usage.html">General usage section</a> is not a big issue.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#generate a variant BED with 3 variants on chr22</span>
<span class="token builtin class-name">echo</span> -e <span class="token string">&quot;chr22<span class="token entity" title="\t">\t</span>15000000<span class="token entity" title="\t">\t</span>16000000<span class="token entity" title="\t">\t</span>deletion<span class="token entity" title="\t">\t</span>None<span class="token entity" title="\t">\t</span>0<span class="token entity" title="\n">\n</span>chr22<span class="token entity" title="\t">\t</span>20000000<span class="token entity" title="\t">\t</span>21000000<span class="token entity" title="\t">\t</span>inversion<span class="token entity" title="\t">\t</span>None<span class="token entity" title="\t">\t</span>0<span class="token entity" title="\n">\n</span>chr22<span class="token entity" title="\t">\t</span>30000000<span class="token entity" title="\t">\t</span>31000000<span class="token entity" title="\t">\t</span>tandem duplication<span class="token entity" title="\t">\t</span>2<span class="token entity" title="\t">\t</span>0&quot;</span> <span class="token operator">&gt;</span> HACk.h1.bed
<span class="token comment">#then run HACk; we can use chr22.fa as reference, as alterations involve only this chromosome</span>
VISOR HACk -g chr22.fa -b HACk.h1.bed -o hack.1.out
<span class="token comment">#as we gave a single BED to HACk without variants involving other haplotypes, HACk generates a single FASTA haplotype (h1.fa) and its index (h1.fa.fai) in the output folder</span>
<span class="token comment">#we can give to HACk as many BED as many haplotypes we want to create</span>
<span class="token builtin class-name">echo</span> -e <span class="token string">&quot;chr22<span class="token entity" title="\t">\t</span>40000000<span class="token entity" title="\t">\t</span>41000000<span class="token entity" title="\t">\t</span>deletion<span class="token entity" title="\t">\t</span>None<span class="token entity" title="\t">\t</span>0&quot;</span> <span class="token operator">&gt;</span> HACk.h2.bed
<span class="token comment">#then run HACk with a double-BED input</span>
VISOR HACk -g chr22.fa -b HACk.h1.bed HACk.h2.bed -o hack.2.out
<span class="token comment">#2 FASTA haplotypes (h1.fa and h2.fa) and their indexes are stored in the output folder</span>
</code></pre></div><h3 id="automatically-building-hack-bed"><a href="#automatically-building-hack-bed" class="header-anchor">#</a> Automatically building HACk BED</h3> <p>When dealing with hundreds of variants, manually building a BED for HACk can be hard. One can then choose to generate a random variant BED with proportions of events or to adapt variants taken from <a href="http://www.internationalgenome.org" target="_blank" rel="noopener noreferrer">public repositories<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> into a BED properly formatted for HACk.</p> <h4 id="generate-a-random-variant-bed"><a href="#generate-a-random-variant-bed" class="header-anchor">#</a> Generate a random variant BED</h4> <p>A <a href="https://github.com/davidebolo1993/VISOR/blob/master/scripts/randomregion.r" target="_blank" rel="noopener noreferrer">R script<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> capable to build a BED file properly formatted for HACk with a user-defined number of random variants is included in the script folder of <a href="https://github.com/davidebolo1993/VISOR" target="_blank" rel="noopener noreferrer">VISOR<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>. All variants described in the <a href="/visordoc/usage/usage.html">General usage section</a> can be generated using this script.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#this script requires a R version &gt;= 3.5.0. It will try to install the needed packages if not already in your package list</span>
Rscript VISOR/scripts/randomregion.r -h 
<span class="token comment">#randomregion.r requires to know your chromosomes dimensions. We can get these easily. For example, using the index of small.genome.fa.fai</span>
<span class="token function">cut</span> -f1,2 small.reference.fa.fai <span class="token operator">&gt;</span> chrom.dim.tsv
<span class="token comment">#generate 100 non-overlapping random variants on chr21 and chr22, with mean length 200 Kb, choosing from deletion, inversion, inverted tandem duplication, translocation copy-paste and reciprocal translocation, with a certain ratio and excluding regions in exclude.bed</span>
Rscript VISOR/scripts/randomregion.r -d chrom.dim.tsv -n <span class="token number">100</span> -l <span class="token number">200000</span> -x exclude.bed -v <span class="token string">'deletion,inversion,inverted tandem duplication,translocation copy-paste,reciprocal translocation'</span> -r <span class="token string">'40:20:20:10:10'</span> <span class="token operator">|</span> sortBed <span class="token operator">&gt;</span> HACk.random.bed
<span class="token comment">#reciprocal translocations are placed by default on a haplotype different then the one specified with the -i parameter (default to 1 -that is, h1 in the final BED). Other translocations types are placed on the same haplotype. Run HACk with this BED</span>
VISOR HACk -g small.reference.fa -b HACk.random.bed -o hack.3.out
<span class="token comment">#2 FASTA haplotypes (h1.fa and h2.fa) with their indexes are stored in the output folder, as 2 haplotypes are present in BED</span>
</code></pre></div><h4 id="generate-a-variant-bed-from-publicly-available-variant-data"><a href="#generate-a-variant-bed-from-publicly-available-variant-data" class="header-anchor">#</a> Generate a variant BED from publicly available variant data</h4> <p>Following is described a standard procedure one can use to get some variant types from publicly available variant data sets in <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank" rel="noopener noreferrer">VCF/BCF format<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> and convert them to BED properly formatted for HACk.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#First, get phased structural variants in VCF format</span>
<span class="token function">curl</span> -LO http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/supporting/GRCh38_positions/ALL.wgs.integrated_sv_map_v2_GRCh38.20130502.svs.genotypes.vcf.gz
<span class="token function">curl</span> -LO http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/integrated_sv_map/supporting/GRCh38_positions/ALL.wgs.integrated_sv_map_v2_GRCh38.20130502.svs.genotypes.vcf.gz.tbi
<span class="token comment">#subset to a single sample (HG00732, for example). Install bcftools if not in PATH</span>
bcftools view -O b -o HG00732.sv.bcf -s HG00732 -m2 -M2 -c <span class="token number">1</span> -C <span class="token number">1</span> ALL.wgs.integrated_sv_map_v2_GRCh38.20130502.svs.genotypes.vcf.gz
bcftools index HG00732.sv.bcf
<span class="token comment">#get haplotype-specific variants and write to BED</span>
bcftools query -f <span class="token string">'%CHROM\t%POS\t%INFO/END\t%INFO/SVTYPE[\t%SAMPLE=%GT]\n'</span> HG00732.sv.bcf <span class="token operator">|</span> <span class="token function">grep</span> -w <span class="token string">&quot;DEL&quot;</span> <span class="token operator">|</span> <span class="token function">grep</span> <span class="token string">&quot;1|0&quot;</span> <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'OFS=FS=&quot;\t&quot;'</span>'<span class="token punctuation">{</span>if <span class="token punctuation">(</span><span class="token variable">$3</span> -<span class="token variable">$2</span> <span class="token operator">&gt;</span><span class="token number">1000</span><span class="token punctuation">)</span> print <span class="token variable">$1</span>, <span class="token variable">$2</span>, <span class="token variable">$3</span>, <span class="token string">&quot;deletion&quot;</span>, <span class="token string">&quot;None&quot;</span>, <span class="token string">&quot;0&quot;</span><span class="token punctuation">}</span><span class="token string">' | sed '</span>s/^/chr/<span class="token string">' &gt; HACk.phased.h1.bed
bcftools query -f '</span>%CHROM<span class="token punctuation">\</span>t%POS<span class="token punctuation">\</span>t%INFO/END<span class="token punctuation">\</span>t%INFO/SVTYPE<span class="token punctuation">[</span><span class="token punctuation">\</span>t%SAMPLE<span class="token operator">=</span>%GT<span class="token punctuation">]</span><span class="token punctuation">\</span>n<span class="token string">' HG00732.sv.bcf | grep -w &quot;DEL&quot; | grep &quot;0|1&quot; | awk '</span><span class="token assign-left variable">OFS</span><span class="token operator">=</span>FS<span class="token operator">=</span><span class="token string">&quot;<span class="token entity" title="\t">\t</span>&quot;</span><span class="token string">''</span><span class="token punctuation">{</span>if <span class="token punctuation">(</span><span class="token variable">$3</span> -<span class="token variable">$2</span> <span class="token operator">&gt;</span><span class="token number">1000</span><span class="token punctuation">)</span> print <span class="token variable">$1</span>, <span class="token variable">$2</span>, <span class="token variable">$3</span>, <span class="token string">&quot;deletion&quot;</span>, <span class="token string">&quot;None&quot;</span>, <span class="token string">&quot;0&quot;</span><span class="token punctuation">}</span><span class="token string">' | sed '</span>s/^/chr/<span class="token string">' &gt; HACk.phased.h2.bed
bcftools query -f '</span>%CHROM<span class="token punctuation">\</span>t%POS<span class="token punctuation">\</span>t%INFO/END<span class="token punctuation">\</span>t%INFO/SVTYPE<span class="token punctuation">[</span><span class="token punctuation">\</span>t%SAMPLE<span class="token operator">=</span>%GT<span class="token punctuation">]</span><span class="token punctuation">\</span>n<span class="token string">' HG00732.sv.bcf | grep -w &quot;INV&quot; | grep &quot;1|0&quot; | awk '</span><span class="token assign-left variable">OFS</span><span class="token operator">=</span>FS<span class="token operator">=</span><span class="token string">&quot;<span class="token entity" title="\t">\t</span>&quot;</span><span class="token string">''</span><span class="token punctuation">{</span>if <span class="token punctuation">(</span><span class="token variable">$3</span> -<span class="token variable">$2</span> <span class="token operator">&gt;</span><span class="token number">1000</span><span class="token punctuation">)</span> print <span class="token variable">$1</span>, <span class="token variable">$2</span>, <span class="token variable">$3</span>, <span class="token string">&quot;inversion&quot;</span>, <span class="token string">&quot;None&quot;</span>, <span class="token string">&quot;0&quot;</span><span class="token punctuation">}</span><span class="token string">' | sed '</span>s/^/chr/<span class="token string">' &gt;&gt; HACk.phased.h1.bed
bcftools query -f '</span>%CHROM<span class="token punctuation">\</span>t%POS<span class="token punctuation">\</span>t%INFO/END<span class="token punctuation">\</span>t%INFO/SVTYPE<span class="token punctuation">[</span><span class="token punctuation">\</span>t%SAMPLE<span class="token operator">=</span>%GT<span class="token punctuation">]</span><span class="token punctuation">\</span>n<span class="token string">' HG00732.sv.bcf | grep -w &quot;INV&quot; | grep &quot;0|1&quot; | awk '</span><span class="token assign-left variable">OFS</span><span class="token operator">=</span>FS<span class="token operator">=</span><span class="token string">&quot;<span class="token entity" title="\t">\t</span>&quot;</span><span class="token string">''</span><span class="token punctuation">{</span>if <span class="token punctuation">(</span><span class="token variable">$3</span> -<span class="token variable">$2</span> <span class="token operator">&gt;</span><span class="token number">1000</span><span class="token punctuation">)</span> print <span class="token variable">$1</span>, <span class="token variable">$2</span>, <span class="token variable">$3</span>, <span class="token string">&quot;inversion&quot;</span>, <span class="token string">&quot;None&quot;</span>, <span class="token string">&quot;0&quot;</span><span class="token punctuation">}</span><span class="token string">' | sed '</span>s/^/chr/' <span class="token operator">&gt;&gt;</span> HACk.phased.h2.bed
<span class="token comment">#run HACk</span>
VISOR HACk -g GRCh38_full_analysis_set_plus_decoy_hla.fa -b HACk.phased.h1.bed HACk.phased.h2.bed -o hack.4.out
<span class="token comment">#among variants that are found to overlap, only the first (is kept)</span>
<span class="token comment">#2 FASTA haplotypes (h1.fa and h2.fa) with their indexes are stored in the output folder</span>
</code></pre></div><h3 id="insert-single-nucleotide-variants-and-structural-variants-in-haplotypes"><a href="#insert-single-nucleotide-variants-and-structural-variants-in-haplotypes" class="header-anchor">#</a> Insert single nucleotide variants and structural variants in haplotypes</h3> <p>HACk's default behaviour is to skip overlapping variants; thus, while is always possible to insert single nucleotide polimorphisms flanking certain structural variants, is impossible to put them in regions with structural variants already inserted. A procedure capable to overcome HACk behaviour is described below.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#First, get phased single nucleotide polimprhisms in VCF format for chr22</span>
<span class="token function">curl</span> -LO ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20181203_biallelic_SNV/ALL.chr22.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz
<span class="token function">curl</span> -LO ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20181203_biallelic_SNV/ALL.chr22.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz.tbi
<span class="token comment">#subset to a single sample (HG00732, for example)</span>
bcftools view -O b -o HG00732.snp.bcf -s HG00732 -m2 -M2 -c <span class="token number">1</span> -C <span class="token number">1</span> ALL.chr22.shapeit2_integrated_v1a.GRCh38.20181129.phased.vcf.gz
bcftools index HG00732.snp.bcf
<span class="token comment">#get haplotype-specific single nucleotide polimorhpisms and write to BED</span>
bcftools query -f <span class="token string">'%CHROM\t%POS\t%REF\t%ALT[\t%SAMPLE=%GT]\n'</span> HG00732.snp.bcf <span class="token operator">|</span> <span class="token function">grep</span> <span class="token string">&quot;1|0&quot;</span> <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'OFS=FS=&quot;\t&quot;'</span>'<span class="token punctuation">{</span>print <span class="token variable">$1</span>, <span class="token punctuation">(</span><span class="token variable">$2</span> -1<span class="token punctuation">)</span>, <span class="token variable">$2</span>, <span class="token string">&quot;SNP&quot;</span>, <span class="token variable">$4</span>, <span class="token string">&quot;0&quot;</span><span class="token punctuation">}</span><span class="token string">' &gt; HACk.snps.h1.bed
bcftools query -f '</span>%CHROM<span class="token punctuation">\</span>t%POS<span class="token punctuation">\</span>t%REF<span class="token punctuation">\</span>t%ALT<span class="token punctuation">[</span><span class="token punctuation">\</span>t%SAMPLE<span class="token operator">=</span>%GT<span class="token punctuation">]</span><span class="token punctuation">\</span>n<span class="token string">' HG00732.snp.bcf | grep &quot;0|1&quot; | awk '</span><span class="token assign-left variable">OFS</span><span class="token operator">=</span>FS<span class="token operator">=</span><span class="token string">&quot;<span class="token entity" title="\t">\t</span>&quot;</span><span class="token string">''</span><span class="token punctuation">{</span>print <span class="token variable">$1</span>, <span class="token punctuation">(</span><span class="token variable">$2</span> -1<span class="token punctuation">)</span>, <span class="token variable">$2</span>, <span class="token string">&quot;SNP&quot;</span>, <span class="token variable">$4</span>, <span class="token string">&quot;0&quot;</span><span class="token punctuation">}</span>' <span class="token operator">&gt;</span> HACk.snps.h2.bed
<span class="token comment">#generate 2 haplotypes (h1.fa and h2.fa) with single nucleotide polimorphisms</span>
VISOR HACk -g chr22.fa -b HACk.snps.h1.bed HACk.snps.h2.bed -o templateswithsnp
<span class="token comment">#use HACk.h1.bed and HACk.h2.bed generated before to insert some simple structural variant. Insert in haplotype 1</span>
VISOR HACk -g templateswithsnp/h1.fa -b HACk.h1.bed -o haplo1
<span class="token comment">#and in haplotype 2</span>
VISOR HACk -g templateswithsnp/h2.fa -b HACk.h2.bed -o haplo2
<span class="token comment">#re-organize the 2 folders, creating one with 2 FASTA haplotypes (h1.fa and h2.fa)</span>
<span class="token function">mv</span> haplo2/h1.fa haplo2/h2.fa <span class="token operator">&amp;&amp;</span> <span class="token function">mv</span> haplo2/h1.fa.fai haplo2/h2.fa.fai <span class="token operator">&amp;&amp;</span> <span class="token function">mv</span> haplo2/h2.fa* haplo1/ <span class="token operator">&amp;&amp;</span> <span class="token function">mv</span> haplo1 hack.5.out <span class="token operator">&amp;&amp;</span> <span class="token function">rm</span> -r haplo2
</code></pre></div><h2 id="visor-shorts-and-laser"><a href="#visor-shorts-and-laser" class="header-anchor">#</a> VISOR SHORtS and LASeR</h2> <p>As described in the <a href="/visordoc/usage/usage.html">General usage section</a>, SHORtS and LASeR require a BED describing regions to simulate, a reference template FASTA (the same used for HACk) and (at least) a folder containing haplotypes generated with HACk. Generating the required BED is pretty straightforward.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#simulate reads aligning to the entire chr22 from haplotypes in test.5.out. As we inserted some structural variants in the 2 haplotypes, their chromosome sizes can differ</span>
<span class="token function">cut</span> -f1,2 hack.5.out/*.fai chr22.fa.fai <span class="token operator">&gt;</span> haplochroms.dim.tsv
<span class="token comment">#chr22 from haplotype 2 is 1000000 base pairs smaller than the one from haplotype 1. For each chromosome, we get the maximum dimension. This is necessary to calculate accurately the number of reads to simulate for each chromosome</span>
<span class="token function">cat</span> haplochroms.dim.tsv <span class="token operator">|</span> <span class="token function">sort</span>  <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'$2 &gt; maxvals[$1] {lines[$1]=$0; maxvals[$1]=$2} END { for (tag in lines) print lines[tag] }'</span> <span class="token operator">&gt;</span> maxdims.tsv
<span class="token comment">#create a BED to simulate reads from chr22, without coverage fluctuations (that is, capture bias value in 4th column is 100.0) and without normal contamination (that is, purity value in 5th column is 100.0) </span>
<span class="token function">awk</span> <span class="token string">'OFS=FS=&quot;\t&quot;'</span>'<span class="token punctuation">{</span>print <span class="token variable">$1</span>, <span class="token string">&quot;1&quot;</span>, <span class="token variable">$2</span>, <span class="token string">&quot;100.0&quot;</span>, <span class="token string">&quot;100.0&quot;</span><span class="token punctuation">}</span>' maxdims.tsv <span class="token operator">&gt;</span> shorts.laser.simple.bed
<span class="token comment">#multiple entries can of course be specified in the same BED</span>
</code></pre></div><p>Once the requirements are met, one can run SHORtS or LASeR.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#run SHORtS first, using the default read simulation settings. Use low coverage to accelerate the alignment step. Use 7 cores for alignment</span>
VISOR SHORtS -g chr22.fa -s hack.5.out -b shorts.laser.simple.bed -o shorts.1.out --threads <span class="token number">7</span> --coverage <span class="token number">5</span>
<span class="token comment">#run LASeR with the default read simulation settings as well. Tag final BAM by haplotype/clone (HP/CL-tags). Using multiple cores to accelerate both simulation and alignment</span>
VISOR LASeR -g chr22.fa -s hack.5.out -b shorts.laser.simple.bed -o laser.1.out --threads <span class="token number">7</span> --coverage <span class="token number">5</span> --tag
<span class="token comment">#a sorted BAM is stored in each output folder</span>
</code></pre></div><p>Generating results with different purity/capture biases combination can be done by simply looping through different, pre-existent, BED files for SHORtS/LASeR. For example, simulating results with different purity values can be done as below.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#bed 100.0 purity is shorts.laser.simple.bed</span>
<span class="token builtin class-name">set</span> <span class="token variable">$VAR2</span>
<span class="token function">cp</span> shorts.laser.simple.bed shorts.laser.purity100.bed
<span class="token comment">#generate also one with purity 80.0 and one with purity 50.0</span>
<span class="token function">awk</span> <span class="token string">'OFS=FS=&quot;\t&quot;'</span>'<span class="token punctuation">{</span><span class="token variable">$5</span><span class="token operator">=</span><span class="token string">&quot;80.0&quot;</span><span class="token punctuation">;</span>print<span class="token punctuation">}</span><span class="token string">' shorts.laser.purity100.bed &gt; shorts.laser.purity80.bed &amp;&amp; awk '</span>OFS<span class="token operator">=</span>FS<span class="token operator">=</span><span class="token string">&quot;<span class="token entity" title="\t">\t</span>&quot;</span><span class="token string">''</span><span class="token punctuation">{</span><span class="token variable">$5</span><span class="token operator">=</span><span class="token string">&quot;50.0&quot;</span><span class="token punctuation">;</span>print<span class="token punctuation">}</span>' shorts.laser.purity100.bed <span class="token operator">&gt;</span> shorts.laser.purity50.bed
<span class="token assign-left variable">VAR1</span><span class="token operator">=</span><span class="token string">&quot;shorts.laser.purity100.bed shorts.laser.purity80.bed shorts.laser.purity50.bed&quot;</span>
<span class="token assign-left variable">VAR2</span><span class="token operator">=</span><span class="token string">&quot;PUR100 PUR80 PUR50&quot;</span>
<span class="token builtin class-name">set</span> <span class="token variable">$VAR2</span>
<span class="token keyword">for</span> <span class="token for-or-select variable">i</span> <span class="token keyword">in</span> <span class="token variable">$VAR1</span><span class="token punctuation">;</span> <span class="token keyword">do</span>
	VISOR SHORtS -g chr22.fa -s hack.5.out -b <span class="token variable">$i</span> -o <span class="token variable">$1</span> --threads <span class="token number">7</span> --coverage <span class="token number">1</span>
	<span class="token builtin class-name">shift</span>
<span class="token keyword">done</span>
</code></pre></div><h3 id="simulate-heterogeneous-bulk-data-with-shorts-and-laser"><a href="#simulate-heterogeneous-bulk-data-with-shorts-and-laser" class="header-anchor">#</a> Simulate heterogeneous bulk data with SHORtS and LASeR</h3> <p>Both SHORtS and LASeR support simulations of bulk data with multiple sub-clones, which is useful to mimic tumour heterogeneity. Reads coming from different clones have a different CL-tag.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#first sub-clone is hack.5.out (diploid clone, with some variants in both haplotypes); second clone is hack.1.out (aneuploid clone with some variants); third clone is templateswithsnp (diploid reference clone). First, get the maximum chromosomes dimensions from multiple clones, as described above</span>
<span class="token function">cut</span> -f1,2 hack.5.out/*.fai hack.1.out/*.fai templateswithsnp/*.fai chr22.fa.fai <span class="token operator">|</span> <span class="token function">sort</span> <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'$2 &gt; maxvals[$1] {lines[$1]=$0; maxvals[$1]=$2} END { for (tag in lines) print lines[tag] }'</span> <span class="token operator">|</span> <span class="token function">awk</span> <span class="token string">'OFS=FS=&quot;\t&quot;'</span>'<span class="token punctuation">{</span>print <span class="token variable">$1</span>, <span class="token string">&quot;1&quot;</span>, <span class="token variable">$2</span>, <span class="token string">&quot;100.0&quot;</span>, <span class="token string">&quot;100.0&quot;</span><span class="token punctuation">}</span>' <span class="token operator">&gt;</span> shorts.laser.multi.bed
<span class="token comment">#run SHORtS first, with multiple samples as input. Each sample is considered as a sub-clone. Ordered percentages describing the relative abundance of each sub-clone are specified with the --clonefraction parameter</span>
VISOR SHORtS -g chr22.fa -s hack.5.out hack.1.out templateswithsnp -b shorts.laser.multi.bed -o shorts.2.out --threads <span class="token number">7</span> --coverage <span class="token number">5</span> --clonefraction <span class="token number">80.0</span> <span class="token number">10.0</span> <span class="token number">10.0</span> --tag
<span class="token comment">#simulate also with LASeR</span>
VISOR LASeR -g chr22.fa -s hack.5.out hack.1.out templateswithsnp -b shorts.laser.multi.bed -o laser.2.out --threads <span class="token number">7</span> --coverage <span class="token number">5</span> --clonefraction <span class="token number">80.0</span> <span class="token number">10.0</span> <span class="token number">10.0</span> --tag
</code></pre></div><h3 id="simulate-single-cell-strand-seq-data-with-shorts"><a href="#simulate-single-cell-strand-seq-data-with-shorts" class="header-anchor">#</a> Simulate single-cell strand-seq data with SHORtS</h3> <p>SHORtS is suitable to simulate single-cell strand-seq data. Strand-seq is a single-cell template strand sequencing technology with directional genomic libraries that allow a clear distinction between the individual homologs of a chromosome. For each haplotype in the given sample, SHORtS generate 2 BAM (one for each strand, usually called Watson and Crick) with the correct read pairs orientation (F1-R2 for Watson and F2-R1 for Crick). Be also sure to set the coverage to a proper, low, value (~ 0.04X, usually)</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#simulte single-cell strand-seq data, adding 5% noise (read pairs with incorrect orientation) with the --noise parameter. We can use a coverage higher than usual one to better visualize the results after the simulations. Use hack.5.out for which we have a ready-to-use BED for SHORtS</span>
VISOR SHORtS -g chr22.fa -s hack.5.out -b shorts.laser.simple.bed -o shorts.3.out --strandseq --threads <span class="token number">7</span> --coverage <span class="token number">1</span> --noise <span class="token number">5.0</span>
<span class="token comment">#this will create 2 subfolders (h1 for first haplotype, h2 for second ...) with a watson (W) and a crick (C) BAM in each. It is also possible to simulate multiple single-cells, some of them originating from a modified clone (sharing the same alterations) and some others from a reference clone. For instance, simulate 2 cells, half (50%) from a reference clone.</span>
VISOR SHORtS -g chr22.fa -s hack.5.out -b shorts.laser.simple.bed -o shorts.4.out --strandseq --threads <span class="token number">7</span> --coverage <span class="token number">1</span> --noise <span class="token number">5.0</span> --cells <span class="token number">2</span> --refcells <span class="token number">50.0</span>
<span class="token comment">#It is also possible to simulate sister chromatid exchange events for cells, haplotypes and regions specified in a separate BED. For h1.fa of the first sample cell, for example:</span>
<span class="token builtin class-name">echo</span> -e <span class="token string">&quot;chr22<span class="token entity" title="\t">\t</span>48000000<span class="token entity" title="\t">\t</span>50818468<span class="token entity" title="\t">\t</span>sample_cell1<span class="token entity" title="\t">\t</span>haplotype1&quot;</span> <span class="token operator">&gt;</span> sce.c1h1.bed
VISOR SHORtS -g chr22.fa -s hack.5.out -b shorts.laser.simple.bed -o shorts.5.out --strandseq --threads <span class="token number">7</span> --coverage <span class="token number">1</span> --noise <span class="token number">5.0</span> --sce sce.c1h1.bed
<span class="token comment">#For the generated cell, we can eventually merge the W/C haplotypes</span>
samtools merge shorts.5.out/sample_cell1/WC.srt.bam shorts.5.out/sample_cell1/h1/W.srt.bam shorts.5.out/sample_cell1/h2/C.srt.bam <span class="token operator">&amp;&amp;</span> samtools index shorts.5.out/sample_cell1/WC.srt.bam
</code></pre></div><p>A method to <a href="https://github.com/davidebolo1993/VISOR/blob/master/scripts/sscounter.py" target="_blank" rel="noopener noreferrer">visualize<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> events in strand-seq data is included in the script folder of <a href="https://github.com/davidebolo1993/VISOR" target="_blank" rel="noopener noreferrer">VISOR<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#a WC.srt.bam is now in the output folder. Plot an interactive visualization of the 2 strands. As we have just chr22, limit the analysis to this chromosome</span>
python VISOR/scripts/sscounter.py -b shorts.5.out/sample_cell1/WC.srt.bam --chromosome chr22 --binsize <span class="token number">50000</span> -o chr22.html
<span class="token comment">#a chr22.html is stored. Can be opened using the default web browser. For example, using firefox</span>
firefox chr22.html
</code></pre></div><h2 id="visor-xenia-beta"><a href="#visor-xenia-beta" class="header-anchor">#</a> VISOR XENIA (BETA)</h2> <p>As described in the <a href="/visordoc/usage/usage.html">General usage section</a>, XENIA requires one BED describing regions to simulate and a folder containing haplotypes generated with HACk. Once the requirements are met, one can run XENIA to simulate linked reads. Linked reads simulations have been inferred from <a href="https://assets.ctfassets.net/an68im79xiti/6ceYcRzVAc6MaSMeyO0akE/4d9f269143be9e1750a415e1d5aa6762/CG00044_10x_Techical_Note_LinkedReads.pdf" target="_blank" rel="noopener noreferrer">10X Genomics documentation<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#use hack.5.out. FASTQ simulation by GEM can be accelerated using multiple cores. The same BED constructed for SHORtS and LASeR can be used. s</span>
VISOR XENIA -s hack.5.out -b shorts.laser.simple.bed -o xenia.1.out 
<span class="token comment">#a FASTQ pair (R1-R2) for each haplotype (L001-L002) is stored in the output folder</span>
</code></pre></div><p>Output from XENIA is ready to be aligned with 10X aligner <a href="https://support.10xgenomics.com/genome-exome/software/pipelines/latest/what-is-long-ranger" target="_blank" rel="noopener noreferrer">Long Ranger<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a>.</p> <div class="language-bash extra-class"><pre class="language-bash"><code><span class="token comment">#create reference folder for Long Ranger</span>
longranger-2.2.2/longranger mkref chr22.fa
<span class="token comment">#convert BCF file of phased SNPs to VCF, required by Long Ranger</span>
bcftools view -O <span class="token function">v</span> -o HG00732.snp.vcf HG00732.snp.bcf
<span class="token comment">#align using WGS Long Ranger module. HG00732.snp.vcf has to be given as a full path</span>
longranger-2.2.2/longranger wgs --reference refdata-chr22 --fastqs xenia.1.out --precalled HG00732.snp.vcf --somatic --sex f --id longranger.out
<span class="token comment">#browse the generate loupe.loupe file using the Diploid Genome Viewer Loupe that can be downloaded from Long Ranger page (https://support.10xgenomics.com/genome-exome/software/downloads/latest)</span>
<span class="token builtin class-name">export</span> <span class="token assign-left variable">LOUPE_SERVER</span><span class="token operator">=</span>longranger.out/outs/
<span class="token function">sh</span> loupe-2.1.1/start_loupe.sh
<span class="token comment">#select the loupe.loupe file and browse. As we simulated just one chromosome, the number of GEMs detected will be lower than expected for an entire genome</span>
</code></pre></div><div class="custom-block tip"><p class="custom-block-title">TIP</p> <p>Please be aware that XENIA is released as a BETA version and some issues may emerge while running; if so, please open an <a href="https://github.com/davidebolo1993/VISOR/issues" target="_blank" rel="noopener noreferrer">issue<span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a> or get in touch with me at davidebolognini7@gmail.com</p></div></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ←
      <a href="/visordoc/usage/usage.html" class="prev">
        General usage
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/visordoc/assets/js/app.6e82cc75.js" defer></script><script src="/visordoc/assets/js/3.6c37c2a3.js" defer></script><script src="/visordoc/assets/js/11.6a825e0e.js" defer></script>
  </body>
</html>
